microservices-connector/cmd/router/main.go

```diff
@@ -477,6 +479,109 @@ func mcGraphHandler(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+func mcDataHandler(w http.ResponseWriter, r *http.Request) {
+	defaultNode := mcGraph.Spec.Nodes[defaultNodeName]
+	for i := range defaultNode.Steps {
+		step := &defaultNode.Steps[i]
+		if DataPrep == step.StepName {
+			log.Info("Starting execution of step", "stepName", step.StepName)
+			serviceURL := getServiceURLByStepTarget(step, mcGraph.Namespace)
+			log.Info("ServiceURL is", "serviceURL", serviceURL)
+			// Parse the multipart form in the request
+			err := r.ParseMultipartForm(64 << 20) // 64 MB is the default used by ParseMultipartForm
+			if err != nil {
+				http.Error(w, "Failed to parse multipart form", http.StatusBadRequest)
+				return
+			}
+			// Create a buffer to hold the new form data
+			var buf bytes.Buffer
+			writer := multipart.NewWriter(&buf)
+			// Copy all form fields from the original request to the new request
+			for key, values := range r.MultipartForm.Value {
+				for _, value := range values {
+					err := writer.WriteField(key, value)
+					if err != nil {
+						http.Error(w, "Failed to write form field", http.StatusInternalServerError)
+						return
+					}
+				}
+			}
+			// Copy all files from the original request to the new request
+			for key, fileHeaders := range r.MultipartForm.File {
+				for _, fileHeader := range fileHeaders {
+					file, err := fileHeader.Open()
+					if err != nil {
+						http.Error(w, "Failed to open file", http.StatusInternalServerError)
+						return
+					}
+					defer func() {
+						if err := file.Close(); err != nil {
+							log.Error(err, "error closing file")
+						}
+					}()
+					part, err := writer.CreateFormFile(key, fileHeader.Filename)
+					if err != nil {
+						http.Error(w, "Failed to create form file", http.StatusInternalServerError)
+						return
+					}
+					_, err = io.Copy(part, file)
+					if err != nil {
+						http.Error(w, "Failed to copy file", http.StatusInternalServerError)
+						return
+					}
+				}
+			}
+			err = writer.Close()
```

Comment:
add writer into defer() block as well

Reply:
Hi @irisdingbj ,  sure, missing this. Done!

Reply:
Hi @irisdingbj, after validation, `writer.Close()` should be operated before the request is sent out. So we can not put it in defer() block, otherwise, the request can not contain the input information, such file name or link path, etc.

Reply:
And I also add `handleMultipartError` to handle the error occurred before `writer.Close()`